api_tests/preprints/views/test_preprint_detail.py

```diff
@@ -857,9 +857,9 @@ def test_update_conflict_of_interest_statement(self, app, user, preprint, url):
         preprint.save()
         res = app.patch_json_api(url, update_payload, auth=user.auth, expect_errors=True)
 
-        assert res.status_code == 400
-        assert res.json['errors'][0]['detail'] == 'You do not have the ability to edit a conflict of interest while the ' \
-                                                  'has_coi field is set to false or unanswered'
+        assert res.status_code == 200
+        assert res.json['data']['attributes']['conflict_of_interest_statement'] ==\
+               'Owns shares in Closed Science Corporation.'
```

Comment:
This doesn't look right. If the has_coi is false in L856, then I don't think you should be able to set the COI statement. I'm also not a big fan of how this test is setting the has_coi statement in the python code rather than just using the api call to set the has_coi at the same time as setting the statement. I know that was there before, but I think it would make more sense for our purposes to verify that a complete API payload with the change in has_coi is combined with the COI statement would work properly.

That being said, I'm not seeing the cases in here that we were testing for explicitly, where if the json request sets `has_coi = false` and also sets `conflict_of_interest_statement = null` that it works properly. 

I know from looking down later in the code that you've made some modifications to the model to clear out values if the has_coi is set to false, but I think that's the wrong way to go, and we'll get to that when we get to the model stuff.